test-network-function/platform/suite.go

```diff
@@ -287,66 +286,39 @@ func testSysctlConfigsHelper(podName, podNamespace string) {
 	}
 }
 
-func testTainted() {
-	var nodeNames []string
+func testTainted(env *config.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestNonTaintedNodeKernelsIdentifier)
 	ginkgo.It(testID, func() {
 		ginkgo.By("Testing tainted nodes in cluster")
-		ginkgo.By("Should return list of node names")
-		context := common.GetContext()
-		tester := nodenames.NewNodeNames(common.DefaultTimeout, nil)
-		test, err := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
-		gomega.Expect(err).To(gomega.BeNil())
-		testResult, err := test.Run()
-		gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
-		gomega.Expect(err).To(gomega.BeNil())
-		nodeNames = tester.GetNodeNames()
-		gomega.Expect(nodeNames).NotTo(gomega.BeNil())
-		ginkgo.By("Should not have tainted nodes")
-		defer results.RecordResult(identifiers.TestNonTaintedNodeKernelsIdentifier)
-		if len(nodeNames) == 0 {
-			ginkgo.Skip("Can't test tainted nodes when list of nodes is empty. Please check previous tests.")
-		}
+
 		var taintedNodes []string
-		for _, node := range nodeNames {
+		for _, node := range env.Nodes {
```

Comment:
Should be filtered node list.

Reply:
also, other platform tests that also use node list should be updated.

Reply:
Other test requires filtering based on whether the cnf work load is present. Will be done by @hamadise